src/Renci.SshNet/SshCommand.cs

```diff
@@ -64,6 +65,25 @@ public class SshCommand : IDisposable
         public Stream ExtendedOutputStream { get; private set; }
 #pragma warning restore CA1859 // Use concrete types when possible for improved performance
 
+        /// <summary>
+        /// Creates and returns the input stream for the command.
+        /// </summary>
+        /// <returns>
+        /// The stream that can be used to transfer data to the command's input stream.
+        /// </returns>
+ #pragma warning disable CA1859 // Use concrete types when possible for improved performance
+        public Stream CreateInputStream()
+#pragma warning restore CA1859 // Use concrete types when possible for improved performance
+        {
+            if (_channel == null)
+            {
+                throw new InvalidOperationException($"The input stream can be used only after calling BeginExecute and before calling EndExecute.");
+            }
+
+            _inputStream ??= new ChannelInputStream(_channel);
```

Comment:
Perhaps it should throw an error if it's already not-null, instead of silently doing nothing?

Reply:
Right.